packages/ncids-js/src/components/usa-site-alert/usa-site-alert.component.ts

```diff
@@ -0,0 +1,249 @@
+	protected options: USASiteAlertOptions;
+	/** The close button component. */
+	private closeButton!: SiteAlertCloseButton;
+	/** The collapse component. */
+	private collapse!: SiteAlertCollapse;
+	/** Default settings for the component, combines close button and collapse components. */
+	private static optionDefaults: USASiteAlertOptions = {
+		closeable: false,
+		closeAriaLabel: 'Dismiss alert',
+		closeButtonClass: 'usa-alert__nci-button--close',
+		closeEventListenerLabel: 'usa-site-alert:close-button',
+		closeCookiePath: '/',
+		collapseAriaLabel: 'Toggle alert message',
+		collapseButtonClass: 'usa-alert__nci-button--toggle',
+		collapseCookiePath: '/',
+		collapseEventListenerLabel: 'usa-site-alert:content',
+	};
+	/** Map object of the component. */
+	private static _components: Map<HTMLElement, USASiteAlert> = new Map<
+		HTMLElement,
+		USASiteAlert
+	>();
+
+	/**
+	 * Initializes class properties then builds component.
+	 *
+	 * @param element Component being created.
+	 * @param options Optional settings for component generation.
+	 * @protected
+	 */
+	protected constructor(
+		element: HTMLElement,
+		options?: Partial<USASiteAlertOptions>
+	) {
+		this.element = element;
+		this.options = {
+			...USASiteAlert.optionDefaults,
+			...options,
+		};
+		const existingComponent = USASiteAlert._components.get(this.element);
+		if (existingComponent) {
+			existingComponent.unregister();
+		}
+		USASiteAlert._components.set(this.element, this);
+
+		this.initialize();
+	}
+
+	/**
+	 * Instantiates this component of the given element.
+	 *
+	 * @param element Component being created.
+	 * @param options Optional settings for component generation.
+	 */
+	public static create(
+		element: HTMLElement,
+		options?: Partial<USASiteAlertOptions>
+	): USASiteAlert {
+		if (!(element instanceof HTMLElement)) {
+			throw 'Element is not an HTMLElement';
+		}
+		return this._components.get(element) || new this(element, options);
+	}
+
+	/**
+	 * Creates all site alert components in document.
+	 * @internal
+	 */
+	public static createAll(): USASiteAlert[] {
+		const siteAlerts = Array.from(
+			document.getElementsByClassName('usa-site-alert')
+		);
+		return siteAlerts.map((siteAlert) => {
+			const closable = siteAlert.hasAttribute('data-site-alert-closable');
+			return this.create(siteAlert as HTMLElement, {
+				closeable: closable,
+			});
+		});
+	}
+
+	/**
+	 * Auto initializes site alert.
+	 * @internal
+	 */
+	public static autoInit(): void {
+		document.addEventListener('DOMContentLoaded', () => {
+			const siteAlerts = Array.from(
+				document.getElementsByClassName('usa-site-alert')
+			);
+			siteAlerts.forEach((siteAlert) => {
+				const alert = siteAlert as HTMLElement;
+				const closeable =
+					alert.dataset.siteAlertClosable?.toLowerCase() === 'true';
+				USASiteAlert.create(alert as HTMLElement, { closeable });
+			});
+		});
+	}
+
```

Comment:
This should just call createAll.

Reply:
createAll is now called. Site alerts are up on the auto init testing page https://designsystem-dev.cancer.gov/pr-1708/example-site/example-pages/auto-init-example.html